src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -109,22 +132,28 @@ public MessageCacheItem getMessages(Iterator<Locale> fallbackLocalesIter) {
 			if (cacheItem.isExpired())
 				refreshCacheItemTask(cacheItem); // Refresh the cacheItem in a separate thread
 		} else { // Item is not in cache.
-			ProductService ps = new ProductService(dto);
-			Locale locale = Locale.forLanguageTag(dto.getLocale());
-			if (ps.isSupportedLocale(locale) || VIPCfg.getInstance().isPseudo()) {
-				cacheItem = createCacheItem(); // Fetch for the requested locale from data store, create cacheItem and store in cache
-				if (cacheItem.getCachedData().isEmpty())  // Failed to fetch messages for the requested locale
-					cacheItem = getFallbackLocaleMessages(fallbackLocalesIter);
-			} else   // Requested locale is not supported and does not match any supported locales
+			cacheItem = createCacheItem(); // Fetch for the requested locale from data store, create cacheItem and store in cache
+			if (cacheItem.getCachedData().isEmpty())  // Failed to fetch messages for the requested locale
 				cacheItem = getFallbackLocaleMessages(fallbackLocalesIter);
 		}
+
 		return cacheItem;
 	}
 
 	private void doLocaleMatching() {
-		Locale matchedLocale = LocaleUtility.pickupLocaleFromList(new ProductService(dto).getSupportedLocales(), Locale.forLanguageTag(dto.getLocale()));
-		if (matchedLocale != null) // Requested locale matches a supported locale (eg. requested locale "fr_CA matches supported locale "fr")
+		dto.setLocale(LocaleUtility.fmtToMappedLocale(dto.getLocale()).toLanguageTag());
```

Comment:
If this is necessary. Suggest to do this when create dto object.

Reply:
This has been there even before this PR. Let's not change it here.